1_fetch/src/get_nwis_data.R

```diff
@@ -1,48 +1,102 @@
+return(finalDf)
+}
+
+# 1.5st function (is not being used)
+download_nwis_data2 <- function(site_nums, parameterCd = '00010') # c("01432160", "01436690") <- simple vector that works without changing dates
+{
   # create the file names that are needed for download_nwis_site_data
-  # tempdir() creates a temporary directory that is wiped out when you start a new R session; 
+  # tempdir() creates a temporary directory that is wiped out when you start a new R session;           # changed 
   # replace tempdir() with "1_fetch/out" or another desired folder if you want to retain the download
-  download_files <- file.path(tempdir(), paste0('nwis_', site_nums, '_data.csv'))
-  data_out <- data.frame()
+  download_files <- file.path("1_fetch/out", paste0('nwis_', site_nums, '_data.csv'))
+  ?browser()
+  data_out <- data.frame(agency_cd = c(), site_no = c(), dateTime = c(),           # standard to dataRetrieval library
+                         X_00010_00000 = c(), X00010_00000_cd = c(), tz_cd = c())
+  
   # loop through files to download 
-  for (download_file in download_files){
-    download_nwis_site_data(download_file, parameterCd = '00010')
+  plzStart <- "2014-05-01"
+  plzStop  <- "2015-05-01"                # paramters modifiable: dcreasing the expanse of time had no affect on build/fail
+  
+  checkExists(download_files, plzStop)      # used to check if we need to update the or not
+  
+  for (download_file in download_files){                               # set start and end dates in here
+    download_nwis_site_data(download_file, parameterCd =  parameterCd, plzStart, plzStop)
     # read the downloaded data and append it to the existing data.frame
-    these_data <- read_csv(download_file, col_types = 'ccTdcc')
+    
+    these_data <- read_csv(download_file, col_types = 'ccTdcc')                  # set the return inside here
     data_out <- bind_rows(data_out, these_data)
   }
   return(data_out)
 }
 
+# optional function
+checkExsits <- function(download_files, endDate, filepath){
+  for (data in download_files) {
+    these_data <- read_csv(download_file, col_types = 'ccTdcc')    # actually I shouldn't use this
+    if (!dir.exists(data) & !data$dateTime.tail() == endDate) {
+      write.csv(data, filepath)
+    }
+    else 
+      download_nwis_site_data(data, parameterCd = '00010', startDate="2014-05-01", endDate="2015-05-01")
+  } 
+}
+
+
+# 2nd function
 nwis_site_info <- function(fileout, site_data){
   site_no <- unique(site_data$site_no)
   site_info <- dataRetrieval::readNWISsite(site_no)
   write_csv(site_info, fileout)
   return(fileout)
 }
+                   
 
-
-download_nwis_site_data <- function(filepath, parameterCd = '00010', startDate="2014-05-01", endDate="2015-05-01"){
-  
-  # filepaths look something like directory/nwis_01432160_data.csv,
+# 3rd funciton
+download_nwis_site_data <- function(filepath, parameterCd = '00010', startDate="2014-05-01", endDate="2015-05-01")
+  {  # changed this function signature too
```

Comment:
what does function signature mean?

Reply:
I included dates in the function signature to allow calls to modify the start and end dates, these are set to hard coded values currently, but they could be switched to variables.